transcendence-app/src/chat/infrastructure/db/chat-room-message.entity.ts

```diff
@@ -0,0 +1,31 @@
+export enum ChatRoomMessageKeys {
+  ID = '"id"',
+  USER_ID = '"userId"',
+  CHAT_ROOM_ID = '"chatRoomId"',
+  CONTENT = '"content"',
+  CREATED_AT = '"createdAt"',
+}
+
+export interface ChatRoomMessageEntityData {
```

Comment:
Why do you need this model of splitting XXXData from XXXEntity with the same fields, and then building Entity from Data?

Reply:
Same doubt!

Reply:
When doing `SELECT * ...`, the node-pg library returns an array of JSON-like objects https://node-postgres.com/api/result#resultrows-arrayany

If we want to instantiate an entity from an object of the pg.Result.rows[i], we need to use the constructor.

Currently, we don't use the entities in our application (we use them like objects and don't use the constructors). For example, let's take a look at the user entity https://github.com/42AGV/ft_transcendence/blob/6740ec026f3b9e6e153d32a8fa5d50a8b0181b34/transcendence-app/src/user/infrastructure/db/user.entity.ts#L13-L24

The constructor takes around nine parameters, and if we want to create an `UserEntity` instance, we would have to pass them in order.

This PR adds the `ChatRoomMessageWithUserEntity`,  and for this, we need to use the constructor https://github.com/42AGV/ft_transcendence/blob/c463830a0d1a7e3914c087cb826b77234afbf7eb/transcendence-app/src/chat/infrastructure/db/chat-room-message-with-user.entity.ts#L22

In order to type the constructor of an XXXEntity, we create the XXXData interface.

Reply:
Ok, so this is more like a data type workaround :+1: